### PR TITLE
[embedded] Do not emit .swiftinterface files for fragile modules (embedded stdlib, Cxx.swiftmodule)

### PR DESCRIPTION
**Pull request by kubamracek**
Currently, we get warnings like this:

```
[1/13] Generating /Users/kuba/swift-github-main/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/./lib/swift/macosx/Cxx.swiftmodule/arm64-apple-macos.swiftmodule
<unknown>:0: warning: module interfaces are only supported with -enable-library-evolution
<unknown>:0: warning: module interfaces are only supported with -enable-library-evolution
[13/13] Generating /Users/kuba/swift-github-main/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/embedded/Swift.swiftmodule/arm64-apple-macos.swiftmodule
<unknown>:0: warning: module interfaces are only supported with -enable-library-evolution
<unknown>:0: warning: module interfaces are only supported with -enable-library-evolution
```

Let's fix it (by not generating the .swiftinterface for these fragile modules), for both the embedded stdlib, and the Cxx module.

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

File: stdlib/cmake/modules/SwiftSource.cmake [hunk 1]
```diff
@@ -646,7 +646,7 @@ function(_compile_swift_files
     set(sibopt_file "${module_base}.O.sib")
     set(sibgen_file "${module_base}.sibgen")
 
-    if(SWIFT_ENABLE_MODULE_INTERFACES)
+    if(SWIFT_ENABLE_MODULE_INTERFACES AND NOT SWIFTFILE_IS_FRAGILE)
```
Comment (kubamracek) [inline stdlib/cmake/modules/SwiftSource.cmake:649]: Note that Cxx is already passing IS_FRAGILE.

**Comment by kubamracek** [comment]
@swift-ci please test Linux platform

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by eeckstein — APPROVED** [review]

**Review by etcwilde — APPROVED** [review]
I makes me a bit sad that `SWIFT_ENABLE_MODULE_INTERFACES` isn't configurable on a per-library basis so that we have to drill holes like this, but that seems to be par for the course.

**Comment by kubamracek** [comment]
> I makes me a bit sad that SWIFT_ENABLE_MODULE_INTERFACES isn't configurable on a per-library basis so that we have to drill holes like this, but that seems to be par for the course.

Isn't that exactly what "IS_FRAGILE" is? Maybe we could deprecate/remove SWIFT_ENABLE_MODULE_INTERFACES in favor of IS_FRAGILE?

**Review by DougGregor — APPROVED** [review]

**Review by zoecarver — APPROVED** [review]

**Pull request merged**